### ISSUE TITLE
Is a CHANGELOG needed? How to know what's new with each release?

### ISSUE THREAD

**Issue by hisham** [aws-amplify/amplify-cli#450]
** Which Category is your question related to? **
ampliy cli

** Provide additional details e.g. code snippets **
I noticed new releases are getting pushed out on a weekly basis or so for the cli. But it's not clear what is changed with every release. Is there a recommended way of figuring this out? This is often done through a CHANGELOG type file...

**Comment by kaustavghosh06** [comment]
@hisham We maintain a CHANGELOG in every package since we maintain our repo as a monorepo and publish every package in the monorepo to npm independently. We can work on consolidating it and putting it at the top level repo if that's what you're looking for.

**Comment by hisham** [comment]
Yeah I think top level changelog would be easier to view as there are lots of packages.

**Comment by ajhool** [comment]
I went to post this same feature request.

It would be extremely useful to have an organized changelog. The main changelog could simply include a link to the changelogs within each package. The team pushes so many features (nice job!) that it's hard to keep track of the current state.

For instance, when using the API package, I don't recall being able to use the OnDemand capacity setting when I first initialized the API. However, I just checked to see if that feature has been added and I found an issue that pointed to a commit where OnDemand capacity support had been added. It would be GREAT if I could just check a changelog every day/week/version to see what features had been added.

I did find: https://github.com/aws-amplify/amplify-cli/blob/master/packages/amplify-cli/CHANGELOG.md , but it is not quite detailed enough. Ideally, there would be a "bug fixes" and "features added" section for each version

**Comment by ajhool** [comment]
I would also suggest using an amplicy CLI command (`amplify changelog`) to display the changelog. If the user's CLI is old, it could potentially query an endpoint to show changes in more recent versions (ie. if I have amplify-cli version 1.0.0; `amplify changelog` would show the changelog up to 1.0.0 but also show changes in 1.1.0, 1.2.0, etc.

**Comment by zfarrell** [comment]
I also started to post this same feature request. 

I would also agree that having a top level changelog would be easier. For example, i wanted to figure out which version `@key` was released with. Nothing in the root project or `packages/amplify-cli`. I'm sure for those in the know, this is clearly called out in one of the 20+ sub-projects, but for someone who doesn't know (me! :) ) it's really confusing to figure out where to look. I gave up and ended up downgrading amplify-cli to figure it out.
